### ISSUE TITLE
Clarification for password reset

### ISSUE THREAD

**Issue by lukstbit** [code4romania/de-urgenta-android#75]
@aniri Is the backend completely implemented for resetting the password? 

I see the endpoint _/auth/request-reset-password_ which has a correspondence to a screen in the application (02.2) , but this endpoint seem to just send a reset email. 

For screen 02.3 I'm assuming that we should hit endpoint _/auth/reset-password_, but what should we use for _userId_ and _resetToken_ if the previous endpoint doesn't return anything?

Any thoughts on this?

**Comment by aniri** [comment]
@lukstbit It seems the reset email has a reset link in it that has the _userId_ and _resetToken_ as query params. We should check if we can use it as a deep link to the 02.3 screen. Or adjust it accordingly from the backend app :D

 What do you think?

**Comment by lukstbit** [comment]
Yes, we could use that url as a deep link if it has those parameters. The problem with this approach is that we will be competing with other apps handling urls(like web browsers) so the user should be informed in that email that he is to handle the reset link with our app in order to reset his password.

**Comment by aniri** [comment]
good point! @lukstbit thanks for the suggestion!

**Comment by lukstbit** [comment]
@aniri I looked at this and a deep link would work. However, I don't think it is a good experience for the user, it's kind of weird. I think it would be better to have the password reset implemented on the client side. This way the password reset will be similar to most applications/websites and it should be really easy to implement in the web vs complicating the app logic(and duplicating it on the two platforms).

Let me know what you think. If you want to have the password reset in the app I'll make a PR tomorrow or saturday.

**Comment by aniri** [comment]
@lukstbit fair enough :D let's not overcomplicate things. we'll postpone this until after we finish the client part and see if it would be enough :)
